python/pandas.cpp

```diff
@@ -174,7 +228,82 @@ void add_key_value<arrow::BinaryArray>( //
     std::size_t row_idx) {
     auto binary_array = std::static_pointer_cast<arrow::BinaryArray>(array);
     auto value = binary_array->Value(row_idx);
-    jsons += fmt::format("\"{}\":\"{}\",", column_name.data(), std::string_view(value.data(), value.size()));
+    fmt::format_to(std::back_inserter(jsons),
+                   "\"{}\":\"{}\",",
+                   column_name.data(),
+                   std::string_view(value.data(), value.size()));
+}
+
+void update(py_table_collection_t& df, py::object obj) {
+    if (!arrow::py::is_batch(obj.ptr()))
+        throw std::invalid_argument("Expected Arrow Table!");
+
+    if (std::holds_alternative<std::monostate>(df.rows_keys)) {
+        auto keys_range = df.binary.native.keys();
+        auto keys_stream = keys_range.begin();
+        std::vector<ukv_key_t> keys_found;
+        while (!keys_stream.is_end()) {
+            keys_found.insert(keys_found.end(), keys_stream.keys_batch().begin(), keys_stream.keys_batch().end());
+            keys_stream.seek_to_next_batch();
+        }
+        df.rows_keys = keys_found;
+    }
+
+    auto& keys = std::get<std::vector<ukv_key_t>>(df.rows_keys);
+    auto collection = docs_collection_t(df.binary.native.db(), df.binary.native, df.binary.native.txn());
+
+    arrow::Result<std::shared_ptr<arrow::RecordBatch>> maybe_record_batch = arrow::py::unwrap_batch(obj.ptr());
+    std::shared_ptr<arrow::RecordBatch> record_batch = maybe_record_batch.ValueOrDie();
+
+    if (record_batch->num_rows() != keys.size())
+        throw std::invalid_argument("record_batch Rows Count Must Match Keys Count");
+
+    std::size_t column_names_length = 0;
+    for (size_t column_idx = 0; column_idx != record_batch->num_columns(); ++column_idx)
+        column_names_length += std::strlen(record_batch->column_name(column_idx).c_str());
+
+    std::string jsons_to_merge;
+    jsons_to_merge.reserve(record_batch->num_rows() * (column_names_length + (record_batch->num_columns() * 3) + 2));
```

Comment:
This should be done better and should include counting the length of all the strings and scalars